### PR TITLE
Move ``TestDocstringCheckerRaise`` to functional tests

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

I won't do `TestParamDocChecker` in one PR as it is too large. I'll try and find a good solution for it!

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]
Very nice !

File: tests/functional/ext/docparams/raise/missing_raises_doc_Sphinx.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    try:
+        raise RuntimeError("hi")
+    except RuntimeError:
+        pass
+
+    raise NameError("hi")
+
+
+def test_find_missing_sphinx_raises_infer_from_instance(self):  # [missing-raises-doc]
+    """This is a docstring.
```
Comment (Pierre-Sassoulas) [inline tests/functional/ext/docparams/raise/missing_raises_doc_Sphinx.py:90]:
```suggestion
    """This is a sphinx docstring.
```

File: tests/functional/ext/docparams/raise/missing_raises_doc_Sphinx.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    try:
+        fake_func()
+    except RuntimeError:
+        raise
+
+    raise NameError("hi")
+
+
+def test_finds_rethrown_sphinx_multiple_raises(self):  # [missing-raises-doc]
+    """This is a docstring.
```
Comment (Pierre-Sassoulas) [inline tests/functional/ext/docparams/raise/missing_raises_doc_Sphinx.py:64]:
```suggestion
    """This is a sphinx docstring.
```

File: tests/functional/ext/docparams/raise/missing_raises_doc_Sphinx.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+    :raises NameError: Never
+    """
+    my_exception = RuntimeError("hi")
+    raise my_exception
+    raise NameError("hi")  # [unreachable]
+
+
+def test_find_missing_sphinx_raises_infer_from_function(self):  # [missing-raises-doc]
+    """This is a docstring.
```
Comment (Pierre-Sassoulas) [inline tests/functional/ext/docparams/raise/missing_raises_doc_Sphinx.py:100]:
```suggestion
    """This is a sphinx docstring.
```

File: tests/functional/ext/docparams/raise/missing_raises_doc_Google.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    try:
+        fake_func()
+    except RuntimeError:
+        raise
+
+    raise NameError("hi")
+
+
+def test_find_rethrown_google_multiple_raises(self):  # [missing-raises-doc]
+    """This is a docstring.
```
Comment (Pierre-Sassoulas) [inline tests/functional/ext/docparams/raise/missing_raises_doc_Google.py:114]:
```suggestion
    """This is a Google docstring.
```

File: tests/functional/ext/docparams/raise/missing_raises_doc_Google.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    try:
+        fake_func()
+    except (RuntimeError, ValueError):
+        raise
+
+    raise NameError("hi")
+
+
+def test_ignores_caught_google_raises(self):
+    """This is a docstring.
```
Comment (Pierre-Sassoulas) [inline tests/functional/ext/docparams/raise/missing_raises_doc_Google.py:128]:
```suggestion
    """This is a Google docstring.
```

File: tests/functional/ext/docparams/raise/missing_raises_doc_Numpy.py [hunk 6]
```diff
@@ -0,0 +1,131 @@
+"""Tests for missing-raises-doc and missing-raises-type-doc for Numpy style docstrings"""
+# pylint: disable=function-redefined, invalid-name, undefined-variable, missing-function-docstring
+# pylint: disable=unused-argument, try-except-raise, import-outside-toplevel
+
+
+def test_find_missing_numpy_raises(self):  # [missing-raises-doc]
+    """This is a docstring.
```
Comment (Pierre-Sassoulas) [inline tests/functional/ext/docparams/raise/missing_raises_doc_Numpy.py:7]:
```suggestion
    """This is a numpy docstring.
```

File: tests/functional/ext/docparams/raise/missing_raises_doc_Numpy.py [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    ------
+    NameError
+        Never
+    """
+    raise RuntimeError("hi")
+    raise NameError("hi")  # [unreachable]
+
+
+def test_find_all_numpy_raises(self):
+    """This is a docstring.
```
Comment (Pierre-Sassoulas) [inline tests/functional/ext/docparams/raise/missing_raises_doc_Numpy.py:19]:
```suggestion
    """This is a numpy docstring.
```

File: tests/functional/ext/docparams/raise/missing_raises_doc_Numpy.py [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+        Always
+    NameError
+        Never
+    """
+    raise RuntimeError("hi")
+    raise NameError("hi")  # [unreachable]
+
+
+def test_find_rethrown_numpy_raises(self):  # [missing-raises-doc]
+    """This is a docstring.
```
Comment (Pierre-Sassoulas) [inline tests/functional/ext/docparams/raise/missing_raises_doc_Numpy.py:33]:
```suggestion
    """This is a numpy docstring.
```

File: tests/functional/ext/docparams/raise/missing_raises_doc_Numpy.py [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+    try:
+        fake_func()
+    except RuntimeError:
+        raise
+
+    raise NameError("hi")
+
+
+def test_find_rethrown_numpy_multiple_raises(self):  # [missing-raises-doc]
+    """This is a docstring.
```
Comment (Pierre-Sassoulas) [inline tests/functional/ext/docparams/raise/missing_raises_doc_Numpy.py:49]:
```suggestion
    """This is a numpy docstring.
```

**Comment by DanielNoord** [comment]
Ah my bad, I saw the approval and thought I could merge. I'll do a follow up with your review :)

**Comment by Pierre-Sassoulas** [comment]
No it's me I approved too fast 😄 I wish it was possible to still apply the suggestion but I don't think so.

**Pull request merged**